frameworks/Groovy/hot/shows/benchmark.show.groovy

```diff
@@ -0,0 +1,106 @@
+import org.apache.commons.lang.StringEscapeUtils
+
+def mongo = show.dbs.mongo
+
+def generator = new Random ()
+
+def generate = {
+	Math.max(generator.nextInt(10000),1)
+}
+
+def templateHeader = """<!DOCTYPE html><html><head><title>Fortunes</title></head><body><table><tr><th>id</th><th>message</th></tr>"""
+
+def templateTail = """</table></body></html>"""
+
+def validateNumQueries = {
+	Integer numQueries
+
+	if (!it) numQueries = 1
+
+	try {
+		numQueries = it[0] as Integer
+	} catch (e) {
+		numQueries = 1
+	}
+	numQueries = Math.max(1,Math.min(500,numQueries))
+}
+
+rest.get("/json").then {
+	[message:'Hello, World!']
+}
+
+rest.get('/db/mongodb').then {
+	mongo.World.findOne([_id:generate()]).promise()
+}
+
+rest.get('/queries/mongodb').then { req ->
+	
+	def numQueries = validateNumQueries(req?.requestParams?.queries)
+	def deferred = show.Deferred()
+	def promise = deferred.promise()
+	def results = []
+
+	(1..numQueries).each { i ->
+		promise = promise.then { 
+			mongo.World.findOne(_id:generate()).promise()
+		}.then { world ->
+			results << world
+			results
+		}
+	}
+	deferred.resolve()
+	promise
+}
+
+rest.get('/fortunes/mongodb').then {
+
+	mongo.Fortune.find().promise().then { fortunes ->
+		
+		fortunes << [_id:'0.', message:'Additional fortune added at request time.']
+		fortunes.each {
+			it._id = Float.parseFloat(it._id) as Integer
+			if (it.message.contains('<script>'))
```

Comment:
This looks like not a valid HTML escaping. What if message contains HTML tags other than `<script>`?